webpack.common.js

```diff
@@ -1,30 +1,45 @@
 const { ESBuildPlugin, ESBuildMinifyPlugin } = require('esbuild-loader');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 
-module.exports = {
-  resolve: {
-    alias: {
-      'video.js': 'video.js/core',
+module.exports = (env) => {
+  const config = {
+    module: {
+      rules: [
+        {
+          test: /\.js$/,
+          loader: 'esbuild-loader',
+        }
+      ]
     },
-  },
-  module: {
-    rules: [
-      {
-        test: /\.js$/,
-        loader: 'esbuild-loader',
-      }
+    optimization: {
+      minimize: true,
+      minimizer: [new ESBuildMinifyPlugin({
+        target: 'es2015'
+      })],
+    },
+    plugins: [
+      new ESBuildPlugin(),
+      new MiniCssExtractPlugin({
+        filename: 'index.css',
+      }),
     ]
-  },
-  optimization: {
-    minimize: true,
-    minimizer: [new ESBuildMinifyPlugin({
-      target: 'es2015'
-    })],
-  },
-  plugins: [
-    new ESBuildPlugin(),
-    new MiniCssExtractPlugin({
-      filename: 'index.css',
-    }),
-  ]
+  };
+
+  if (!env.vhs) {
+    config.resolve = {
+      alias: {
+        'video.js': 'video.js/core',
+      }
+    };
+  } else {
+    config.resolve = {
+      alias: {
+        'hls.js': false,
```

Comment:
does this mean that this line:

https://github.com/muxinc/videojs-mux-kit/blob/main/src/utils/mux-data-middleware.js#L4

```js
import Hls from "hls.js";
```

will actually **not** import anything?

Reply:
yup, `Hls` will end up being `undefined` in the that package. See https://github.com/muxinc/videojs-mux-kit/pull/63/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R48 for example where I excluded a file in the VHS build